showcase/app/components/page-components/copy/button/sub-sections/demos.gts

```diff
@@ -0,0 +1,276 @@
+import { eq } from 'ember-truth-helpers';
+import { get } from '@ember/object';
+
+import ShwDivider from 'showcase/components/shw/divider';
+import ShwFlex from 'showcase/components/shw/flex';
+import ShwGrid from 'showcase/components/shw/grid';
+import ShwTextH2 from 'showcase/components/shw/text/h2';
+import ShwTextH3 from 'showcase/components/shw/text/h3';
+import ShwTextH4 from 'showcase/components/shw/text/h4';
+
+import {
+  HdsCopyButton,
+  HdsDropdown,
+} from '@hashicorp/design-system-components/components';
+
+import CodeFragmentWithMaskedInput, {
+  MASKED_INPUT_VARIANTS,
+} from 'showcase/components/page-components/copy/button/code-fragments/with-masked-input';
+import type { CodeFragmentWithMaskedInputSignature } from 'showcase/components/page-components/copy/button/code-fragments/with-masked-input';
+import CodeFragmentWithHdsInput, {
+  INPUT_COMPONENTS,
+} from 'showcase/components/page-components/copy/button/code-fragments/with-hds-input';
+import CodeFragmentWithModal from 'showcase/components/page-components/copy/button/code-fragments/with-modal';
+import CodeFragmentWithGenericDialogContent from 'showcase/components/page-components/copy/button/code-fragments/with-generic-dialog-content';
+import CodeFragmentWithHtmlInput, {
+  INPUT_COMPONENTS as HTML_INPUT_COMPONENTS,
+} from 'showcase/components/page-components/copy/button/code-fragments/with-html-input';
+
+const maskedInputVariantToLabelMap: Record<
+  CodeFragmentWithMaskedInputSignature['Args']['variant'],
+  string
+> = {
+  'masked-input-base': 'With MaskedInputBase',
+  'masked-input-base-form-field': 'With FormField + MaskedInputBase',
+  'masked-input-field': 'With MaskedInputField',
+};
+
+export default class SubSectionDemos extends Component {
+  bigIntValue = BigInt(12345678910);
+
+  get targetNodeElement() {
+    const element = document.querySelector('#test-target-node-element');
+    if (!element) {
+      return undefined;
+    }
+
+    return element as HTMLElement;
+  }
+
+  // DEBUG
+  // uncomment these if you need to debug the `onSuccess/onError` callback methods
+
+  // onSucces = () => ({
+  //   trigger,
+  //   text,
+  //   target,
+  // }: {
+  //   trigger: HTMLElement;
+  //   text: string;
+  //   target: string;
+  // }) {
+  //   console.log('onSuccess invoked in the controller', trigger, text, target);
+  // }
+
+  // onError = () => ({
+  //   trigger,
+  //   text,
+  //   target,
+  // }: {
+  //   trigger: HTMLElement;
+  //   text: string;
+  //   target: string;
+  // }) {
+  //   console.log('onError invoked in the controller', trigger, text, target);
+  // }
+
+  <template>
+    <ShwTextH2>Compositions</ShwTextH2>
+
+    {{#each MASKED_INPUT_VARIANTS as |variant|}}
+      <ShwFlex as |SF|>
+        {{#let (array false true) as |isMultilineOptions|}}
+          {{#each isMultilineOptions as |isMultiline|}}
+            <SF.Item @label={{get maskedInputVariantToLabelMap variant}}>
+              <CodeFragmentWithMaskedInput
```

Comment:
[question] from what I see, `<CodeFragmentWithMaskedInput>` has only this instance. Why then have a code fragment for it (which comes with some extra code, like the `maskedInputVariantToLabelMap` above)? maybe the "Compositions" could be its own sub-section, and contain all the logic related to these examples?

Reply:
Totally fair, I removed this code fragment.